src/vm/i386/unixstubs.cpp

```diff
@@ -6,17 +6,17 @@
 
 extern "C"
 {
-    void ProfileEnterNaked(FunctionIDOrClientID functionIDOrClientID)    
+    void __stdcall ProfileEnterNaked(FunctionIDOrClientID functionIDOrClientID)
```

Comment:
Is the JIT on Linux x86 going to call these with __stdcall calling convention? I thought we have converted everything to cdecl.

Reply:
This one appears documented as though it was __stdcall: https://github.com/dotnet/coreclr/blob/master/src/jit/codegencommon.cpp#L6723
and from my admitedly limited understanding of the JIT code that is how its currently implemented.

@dotnet/jit-contrib - does that sound right?

Assuming it is, LGTM.

Reply:
We have in file src/vm/eetoprofinterfaceimpl.cc
```
// Declarations for asm wrappers of profiler callbacks
EXTERN_C void __stdcall ProfileEnterNaked(FunctionIDOrClientID functionIDOrClientID);
EXTERN_C void __stdcall ProfileLeaveNaked(FunctionIDOrClientID functionIDOrClientID);
EXTERN_C void __stdcall ProfileTailcallNaked(FunctionIDOrClientID functionIDOrClientID);
```
so, when we try to profile now memory usage in optimized version of coreclr we get inconsistent stack after calling one of this stub.

Reply:
Would it be better to fix this in the JIT to use cdecl, so that we are using cdecl across the board on Linux x86? Or is there a good reason why this should be an outlier on Linux x86?

Reply:
@jkotas Do you mean that I need to add ```#ifdef x86 && linux``` with ```cdecl``` declarations in the file src/vm/eetoprofinterfaceimpl.cc?

Reply:
You can delete the `stdcall` there and let it use the default (ie stdcall on Windows and cdecl on Linux). It is what we was done in number of places during Linux x86 bring up.

The key place to get in sync is the JIT though.

Reply:
I don't know what the rationale was to use cdecl on Linux so I don't know what advantage we'd get by switching it (other than the knowledge we are consistent). From the standpoint of codesize, changing this to cdecl would require adding a pop instruction to the prologue of every managed method so probably a net performance loss.

Reply:
There was a long discussion and stream of fixes about this during the Linux x86 bring up. Here are a few related PRs:

https://github.com/dotnet/coreclr/pull/9928
https://github.com/dotnet/coreclr/pull/9977
https://github.com/dotnet/coreclr/pull/10410

The summary is:

- ESP is 16-byte aligned on Linux x86. It makes use of stdcall awkward and inefficient. For example, when we were still using stdcall, the JIT ended up generating code like this:

```
sub esp, 12 // necessary to maintain stack alignment
push <argument>
call method
add esp, 12
```

- cdecl is the native calling convention on Linux x86. We have found that stdcall is handled poorly throughout the system. For example, unwinders do not understand, the C/C++ compiler does not generate good code for it - it calls the method as cdecl and then compensates for callee poped arguments like this: 

```
call method_with_stdcall_convention
sub esp, <size of arguments>
```

It is true that stdcall code w/o the stack alignment requirement can be smaller. However, stdcall is one of the reasons why x86 is different from most other platforms out there. I guess that the Linux folks decided long time ago that it is not worth the on-going pain, it is better to take a small hit and just use cdecl for simplicity.

We have not optimized the JIT for cdecl. The focus of Linux x86 port has been functionality, not performance so far. The JIT does not generate as good code as it can on Linux x86. https://github.com/dotnet/coreclr/issues/10012 is the key part of that.